### PR TITLE
Hook up new event model to EventDetails Screen

### PR DESCRIPTION
**Pull request by RGBboy**
Follow up PR to #376 in order to switch the EventDetails screen to use `state.data.events` to get event details. Slowly getting closer to removing the deprecated events and entries.

This PR also adds the initial [reselect](https://github.com/reduxjs/reselect) and starts to refactor our selectors so they can be more easily composed with reselect. Once this pattern is applied to the Event Listing Screen we should see a performance boost as reselect memoizes selectors.

Unfortunately I had to update the way test data is calculated and therefore a number of snapshots updated. I have double checked them but it might be a good idea to get another pair of eyes.

### CODE REVIEW

File: src/screens/EventDetailsScreen/EventOverview.js [hunk 1]
```diff
@@ -63,32 +61,30 @@ const EventOverview = ({ event }: Props) => {
...
           {formatLongEventPrice(
-            event.fields.eventPriceLow[locale],
-            event.fields.eventPriceHigh[locale]
+            event.fields.eventPriceLow,
+            event.fields.eventPriceHigh
           )}
         </Text>
       </IconItem>
 
       {event.fields.venueDetails &&
```
Comment (frigus02) [inline src/screens/EventDetailsScreen/EventOverview.js]: In the new event type `event.fields.venueDetails` is always an array, so we should be able to remove this check. Similar to `accessibilityOptions`.

File: src/screens/EventDetailsScreen/component.js [hunk 2]
```diff
@@ -19,10 +20,10 @@ import LayoutColumn from "../../components/LayoutColumn";
 import ShadowedScrollView from "../../components/ShadowedScrollView";
 import SectionDivider from "../../components/SectionDivider";
 import PerformanceList from "../../components/PerformanceList";
-import { groupPerformancesByPeriod } from "../../selectors/events";
+// Move this to performance selector
```
Comment (frigus02) [inline src/screens/EventDetailsScreen/component.js]: It is already in the performance selectors, isn't it?

File: src/screens/EventDetailsScreen/component.js [hunk 3]
```diff
@@ -166,20 +163,16 @@ class EventDetailsScreen extends PureComponent<Props> {
...
                       {(event.fields.email || event.fields.phone) && (
                         <EventContact
-                          email={
-                            event.fields.email && event.fields.email[locale]
-                          }
-                          phone={
-                            event.fields.phone && event.fields.phone[locale]
-                          }
+                          email={event.fields.email && event.fields.email}
+                          phone={event.fields.phone && event.fields.phone}
```
Comment (frigus02) [inline src/screens/EventDetailsScreen/component.js]: We should be able to remove these checks and just pass down `event.fields.email` and `event.fields.phone`.

File: src/screens/EventDetailsScreen/component.js (hunk 2 above)
Comment (RGBboy) [reply src/screens/EventDetailsScreen/component.js]: Yep, originally it wasn't. This was a note for me to do that work that I forgot to remove

File: src/components/__snapshots__/Image.test.js.snap [hunk 4]
```diff
@@ -7,7 +7,7 @@ exports[`renders correctly 1`] = `
   source={
     Object {
       "height": 357,
-      "id": "3CU8n6O760GJ80Wr7r",
+      "id": "4D776SGAHYRnNGHx1DES622EFYZ",
```
Comment (markholland) [inline src/components/__snapshots__/Image.test.js.snap]: If we update to Jest 23 we can avoid these sorts of snapshot changes where the value is unimportant by using [property matchers](https://facebook.github.io/jest/blog/2018/05/29/jest-23-blazing-fast-delightful-testing.html#snapshot-property-matchers)

Comment (RGBboy) [reply src/components/__snapshots__/Image.test.js.snap]: Interesting. I will take a look. Do you think it is worthwhile looking now or as something after this refactor?

Comment (markholland) [reply src/components/__snapshots__/Image.test.js.snap]: It shouldn't block this but definitely check it out.

**Pull request merged**